### ISSUE TITLE
Linux wrapper

### ISSUE THREAD

**Issue by sw-libelium** [jimfred/PyCubeProgrammer_API#1]
Hi Jimfred

Currently, I'm trying to fit your project for Linux SO. The problem is that the .dlls files from ST cubeprogrammer do not work directly on Linux.

In addition, seems that in api floder there is no elf files or similar. How would you do it?

Thanks in advance

**Comment by jimfred** [comment]
Good idea. Looking at https://www.st.com/en/development-tools/stm32cubeprog.html#get-software, there's a download called 'STM32CubePrg-Lin | STM32CubeProgrammer software for Linux' which I'd expect would have all the drivers, libraries and scripts required to program a STM32 target using an ST-Link. The Linux CubeIde of course has a debugger that works on Linux. I don't know if there are Linux code examples that utilize that interface but that's a place to start. I've always been curious about ST's software distribution for ST-Link apps and how they get it to work on multiple platforms - I just haven't looked yet.
